packages/browser-wallet/src/popup/pages/Account/TransactionList/TransactionList.tsx

```diff
@@ -0,0 +1,204 @@
+                                itemCount={headersAndTransactions.length}
+                                onItemsRendered={onItemsRendered}
+                                ref={ref}
+                                width={width}
+                                height={height}
+                                itemSize={(i) =>
+                                    isHeader(headersAndTransactions[i])
+                                        ? transactionHeaderHeight
+                                        : transactionElementHeight
+                                }
+                                itemKey={(i) => getKey(headersAndTransactions[i])}
+                                innerElementType={ListElement}
+                            >
+                                {({ index, style }) => {
+                                    const item = headersAndTransactions[index];
+
+                                    if (!isItemLoaded(index)) {
+                                        return <div style={style}>Loading</div>;
+                                    }
+
+                                    if (isHeader(item)) {
+                                        return (
+                                            <span
+                                                style={style}
+                                                className="transaction-list__scroll__transaction-group-header-placeholder"
+                                            />
+                                        );
+                                    }
+                                    return (
+                                        <TransactionElement
+                                            accountAddress={accountAddress}
+                                            style={style}
+                                            key={item.transactionHash ?? item.id}
+                                            transaction={item}
+                                        />
+                                    );
+                                }}
+                            </List>
+                        )}
+                    </InfiniteLoader>
+                )}
+            </AutoSizer>
+        </StickyContext.Provider>
+    );
+}
+
+/**
+ * Displays a list of transactions from an account's transaction history.
+ */
+export default function TransactionList() {
+    const accountAddress = useAtomValue(selectedAccountAtom);
+    const [transactions, setTransactions] = useState<BrowserWalletTransaction[]>([]);
+    const [isNextPageLoading, setIsNextPageLoading] = useState<boolean>(false);
+    const [hasNextPage, setHasNextPage] = useState<boolean>(true);
+
+    if (!accountAddress) {
+        return null;
+    }
+
+    async function loadTransactionsDescending(address: string, appendTransactions: boolean, fromId?: number) {
+        setIsNextPageLoading(true);
+        getTransactions(address, 20, 'descending', fromId)
```

Comment:
is 20 a little low?
Also this constant should not be defined here.

Reply:
The view can at most show 5 transactions at a time, so I thought loading 4x one "view" was close to what we do in the desktop wallet. I will keep it at 20 for now, it is easy to tune at a later point.